src/treeherder/job_handler.js

```diff
@@ -71,26 +71,59 @@ const EVENT_MAP = {
   [events.taskException().exchange]: 'exception'
 };
 
+/**
+ * Parses a task route and inspects task.extra.treeherder for revision information.
+ * Attempt to use task.extra.treeherder.revision if it exists.  If not,
+ * use whatever hash is found in the routing key.
+ * Note: This is meant as a transitional period between
+ * using task.extra and relying soley on the routing key.  Routing
+ * key in the future will be the preferred and only method.
+ */
 export function parseTaskRevisionHash(task, prefix) {
-  // If task.extra.revision and task.extra.revision_hash exist, use those
-  // for submitting treeherder jobs instead of parsing revision hash from
-  // the route
-  if (task.extra && task.extra.treeherder) {
-    let treeherder = task.extra.treeherder
-    if (treeherder.revision && treeherder.revision_hash) {
-      return [treeherder.revision, treeherder.revision_hash];
-    }
+  let revision = _.get(task, 'extra.treeherder.revision');
+  let revision_hash = _.get(task, 'extra.treeherder.revision_hash');
+
+  if (revision && revision_hash) {
+    return [revision, revision_hash];
```

Comment:
I'm wondering if we need to have both here.  If you only have the ``revision`` then you could still just return ``[rev, ""]``